RecoMET/METAlgorithms/interface/EcalHaloAlgo.h

```diff
@@ -91,6 +103,10 @@ class EcalHaloAlgo
   // Phi Wedge Thresholds
   float SumEnergyThreshold;
   int NHitsThreshold;
+
+  const CaloGeometry *geo;
```

Comment:
It would be much safer not to cache geometry in the data members and instead just pass it to the member methods as an argument.